pkg/controller/keycloak/keycloak_reconciler_test.go

```diff
@@ -375,6 +376,109 @@ func TestKeycloakReconciler_Test_Updating_External_Database_URI(t *testing.T) {
+	keycloakSpec := desiredState[5].(common.GenericCreateAction).Ref.(*v13.StatefulSet).Spec.Template.Spec
+
+	// then
+	envVarOk := false
+	for _, a := range keycloakSpec.Containers[0].Env {
+		if a.Name == model.KeycloakDatabaseConnectionParamsProperty && strings.Contains(a.Value, "sslmode=required") {
+			envVarOk = true
+		}
+	}
+	assert.True(t, envVarOk)
+
+	sslVolumeExists := false
+	for _, volume := range keycloakSpec.Volumes {
+		if strings.Contains(volume.Name, model.DatabaseSecretSslCert+"-vol") {
+			sslVolumeExists = true
+		}
+	}
+	assert.True(t, sslVolumeExists)
+}
+
+func TestKeycloakReconciler_Test_Given_SSLMODE_And_RHSSO_When_Reconcile_Then_NewEnvVarsAndMountedVolume(t *testing.T) {
+	// given
+	cr := &v1alpha1.Keycloak{}
+	cr.Spec.ExternalDatabase.Enabled = true
+	cr.Spec.Profile = "RHSSO"
+
+	currentState := common.NewClusterState()
+	currentState.DatabaseSecret = model.DatabaseSecret(cr)
+	currentState.DatabaseSecret.Data[model.DatabaseSecretSslModeProperty] = []byte("required")
+	currentState.DatabaseSSLSecret = model.DatabaseSecret(cr)
+
+	// when
+	reconciler := NewKeycloakReconciler()
+	desiredState := reconciler.Reconcile(currentState, cr)
+	// element 5 is the KeycloakDeployment
+	keycloakSpec := desiredState[5].(common.GenericCreateAction).Ref.(*v13.StatefulSet).Spec.Template.Spec
+
+	// then
+	envVarFound := 0
+	for _, a := range keycloakSpec.Containers[0].Env {
+		if strings.Contains(a.Name, model.RhssoDatabaseNONXAConnectionParamsProperty) && strings.EqualFold(a.Value, "required") {
+			envVarFound++
+		} else if strings.Contains(a.Name, model.RhssoDatabaseXAConnectionParamsProperty) && strings.EqualFold(a.Value, "required") {
+			envVarFound++
+		}
+	}
+	assert.Equal(t, 2, envVarFound)
+
+	sslVolumeExists := false
+	for _, volume := range keycloakSpec.Volumes {
+		if strings.Contains(volume.Name, model.DatabaseSecretSslCert+"-vol") {
+			sslVolumeExists = true
+		}
+	}
+	assert.True(t, sslVolumeExists)
+}
+
+func TestKeycloakReconciler_Test_Given_NoSSLMODE_When_Reconcile_Then_NoNewEnvVarsAndMountedVolume(t *testing.T) {
+	// given
+	cr := &v1alpha1.Keycloak{}
+	cr.Spec.ExternalDatabase.Enabled = true
+
+	currentState := common.NewClusterState()
+	currentState.DatabaseSecret = model.DatabaseSecret(cr)
+
+	// when
+	reconciler := NewKeycloakReconciler()
+	desiredState := reconciler.Reconcile(currentState, cr)
+	// element 5 is the KeycloakDeployment
+	keycloakSpec := desiredState[5].(common.GenericCreateAction).Ref.(*v13.StatefulSet).Spec.Template.Spec
+
+	// then
+	sslVolumeExists := false
+	for _, volume := range keycloakSpec.Volumes {
+		if strings.Contains(volume.Name, model.DatabaseSecretSslCert+"-vol") {
+			sslVolumeExists = true
+		}
+	}
+	assert.False(t, sslVolumeExists)
+
+	envVarOk := false
+	for _, a := range keycloakSpec.Containers[0].Env {
+		if a.Name == model.KeycloakDatabaseConnectionParamsProperty && strings.Contains(a.Value, "sslmode") {
+			envVarOk = true
+		}
+	}
+	assert.False(t, envVarOk)
```

Comment:
I believe you don't need to use a `for` loops which is a bit cumbersome. You can do something like [this](https://stackoverflow.com/a/2050629) instead. But it's just a nitpick – I'll leave it up to you if you want to change it. ;)